catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -155,18 +212,22 @@ private void fetchNextResults() {
             queryCopy.setStartIndex(currentIndex);
 
             try {
-                SourceResponse response = queryFunction.query(queryRequestCopy);
+                response = queryFunction.query(queryRequestCopy);
 
                 if (response.getResults()
                         .size() == 0) {
-                    results = RESULTS_EXHAUSTED;
+                    finished = true;
                     return;
                 }
 
                 results = response.getResults()
                         .iterator();
                 currentIndex += response.getResults()
                         .size();
+
```

Comment:
❓How about `(response.getHits() => 0) && (currentIndex > response.getHits())`? Also, should we always set the `requestsTotalResultsCount` flag here https://github.com/codice/ddf/pull/2207/files#diff-9ee238e3d9003aab5d04411cde4ca2e6R245 to `true`?